src/libsystemd/sd-device/sd-device.c

```diff
@@ -1239,10 +1239,7 @@ static int handle_db_line(sd_device *device, char key, const char *value) {
 
                 break;
         case 'W':
-                r = safe_atoi(value, &device->watch_handle);
-                if (r < 0)
-                        return r;
-
+                /* Deprecated. */
```

Comment:
what happens during upgrade from old udev?

Reply:
Watch handle is always read from /run/udev/watch now. New udevd simply ignores the handle saved in udev databse.